### ISSUE TITLE
how to maintain a radix tree in god?

### ISSUE THREAD

**Issue by firozkhan0910** [zond/god#22]
we want to maintain a radix tree in god db. is it possible?
rather than putting it as a sub-tree under a node, we want to use the internal radix tree of god. so we just insert key/value into god and god maintains the radix tree for us. also prefix based search is desired.

**Comment by zond** [comment]
What you say should work :)

**Comment by firozkhan0910** [comment]
Hi,
thnx for the quick reply. I was able to setup the code base and try some API functions.
I have some questions -

how we can disable the hashing of the keys? we just want the keys to remain in plain.

I could not find a way to do prefix based search. Is it possible to search the database for a set of keys starting with a common prefix.

**Comment by zond** [comment]
The keys aren't hashed, IIRC, it's up to the client to hash them if it
wants to.

https://godoc.org/github.com/zond/god/client#Conn.Next gives you the next
single key after another key. Batch "next" isn't implemented AFAIK.


On Apr 18, 2017 02:41, "firozkhan0910" <notifications@github.com> wrote:

Hi,
thnx for the quick reply. I was able to setup the code base and try some
API functions.
I have some questions -

how we can disable the hashing of the keys? we just want the keys to remain
in plain.

I could not find a way to do prefix based search. Is it possible to search
the database for a set of keys starting with a common prefix.

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
<https://github.com/zond/god/issues/22#issuecomment-294637938>, or mute the
thread

**Comment by firozkhan0910** [comment]
Hi Zond,
If I want to (update the code base) to do a batch next and find out an empty node, how can I achieve this? Plz provide some direction around this.

**Comment by zond** [comment]
Look at how the `Next` API works, and build something like it (unfortunately `Slice` is already taken) that picks everything from `A` to `B` from the node owning `A`, then let the client handle picking out the last element it got and doing the same command again with that element as `A` (and thus getting everything from `A'` to `B`). That's probably easiest.

**Comment by firozkhan0910** [comment]
Hi Zond,
tried my best to digest, request you to plz translate the answer above.
plz share some pseudo-code if possible.

 I'll try to explain the required behavior through an example -

the length of the keys is constant. let's say the tree has following two keys 496123757160001 and 496123757160009. we need to find all the nodes below 49612375716000. in this case we should get two nodes representing the last digit 1 and 9. with this the next available node would be 2 and this way we'll insert 496123757160002 in the tree. if all the leaf nodes at last last digit (1 through 9) are occupied we go one level up and search with sub key 4961237571600, and so on.

**Comment by zond** [comment]
Yeah, this is exactly what `Next` and `Prev` does, but one key at a time.

To implement a batch loader that loads lots of data between two keys you will have to read and understand the code for `Next`, then add a new function that does the same but returns with all following values up to your end key.
